### PR TITLE
Make "AWS_DEFAULT_REGION" required and add to docs

### LINKED ISSUE #1228: [BUG] - error deploying QHub on AWS on 0.4
### OS system and architecture in which you are running QHub

Ubuntu, x86_64

### Expected behavior

QHub deploys a new cluster

### Actual behavior

Qhub failed deployment at stage 3

### How to Reproduce the problem?

Conda installed the latest qhub. Verified I got the latest release:
```
$ qhub --version
0.4.0
```

Initiated qhub with `qhub init aws --project qhub-demo --ci-provider github-actions`

Deployed with `qhub deploy -c qhub-config.yaml`

Failed the first time, and then I changed the project name in qhub-config.yaml to be `qhubdemo` supposing the extra dash was causing the error. No luck.

Below is my qhub config

```
project_name: qhubdemo
provider: aws
domain: qhubtest.cloud
certificate:
  type: self-signed
security:
  authentication:
    type: GitHub
    config:
      client_id: 5fefee...
      client_secret: 4qgf...
  keycloak:
    initial_root_password: FcpW...
default_images:
  jupyterhub: quansight/qhub-jupyterhub:v0.4.0
  jupyterlab: quansight/qhub-jupyterlab:v0.4.0
  dask_worker: quansight/qhub-dask-worker:v0.4.0
  dask_gateway: quansight/qhub-dask-gateway:v0.4.0
storage:
  conda_store: 60Gi
  shared_filesystem: 100Gi
theme:
  jupyterhub:
    hub_title: QHub - qhub-demo
    hub_subtitle: Autoscaling Compute Environment on Amazon Web Services
    welcome: Welcome to None. It is maintained by <a href="http://quansight.com">Quansight
      staff</a>. The hub's configuration is stored in a github repository based on
      <a href="https://github.com/Quansight/qhub/">https://github.com/Quansight/qhub/</a>.
      To provide feedback and report any technical problems, please use the <a href="https://github.com/Quansight/qhub/issues">github
      issue tracker</a>.
    logo: /hub/custom/images/jupyter_qhub_logo.svg
    primary_color: '#4f4173'
    secondary_color: '#957da6'
    accent_color: '#32C574'
    text_color: '#111111'
    h1_color: '#652e8e'
    h2_color: '#652e8e'
helm_extensions: []
monitoring:
  enabled: true
cdsdashboards:
  enabled: true
  cds_hide_user_named_servers: true
  cds_hide_user_dashboard_servers: false
ci_cd:
  type: github-actions
  branch: main
  commit_render: true
terraform_state:
  type: remote
namespace: dev
qhub_version: 0.4.0
amazon_web_services:
  region: us-west-2
  kubernetes_version: '1.22'
  node_groups:
    general:
      instance: m5.xlarge
      min_nodes: 1
      max_nodes: 1
    user:
      instance: m5.large
      min_nodes: 1
      max_nodes: 5
    worker:
      instance: m5.large
      min_nodes: 1
      max_nodes: 5
profiles:
  jupyterlab:
  - display_name: Small Instance
    description: Stable environment with 1 cpu / 4 GB ram
    default: true
    kubespawner_override:
      cpu_limit: 1
      cpu_guarantee: 0.75
      mem_limit: 4G
      mem_guarantee: 2.5G
  - display_name: Medium Instance
    description: Stable environment with 2 cpu / 8 GB ram
    kubespawner_override:
      cpu_limit: 2
      cpu_guarantee: 1.5
      mem_limit: 8G
      mem_guarantee: 5G
  dask_worker:
    Small Worker:
      worker_cores_limit: 1
      worker_cores: 0.75
      worker_memory_limit: 4G
      worker_memory: 2.5G
      worker_threads: 1
    Medium Worker:
      worker_cores_limit: 2
      worker_cores: 1.5
      worker_memory_limit: 8G
      worker_memory: 5G
      worker_threads: 2
environments:
  environment-dask.yaml:
    name: dask
    channels:
    - conda-forge
    dependencies:
    - python
    - ipykernel
    - ipywidgets
    - qhub-dask ==0.3.13
    - python-graphviz
    - numpy
    - numba
    - pandas
  environment-dashboard.yaml:
    name: dashboard
    channels:
    - conda-forge
    dependencies:
    - python==3.9.7
    - ipykernel==6.4.1
    - ipywidgets==7.6.5
    - qhub-dask==0.3.13
    - param==1.11.1
    - python-graphviz==0.17
    - matplotlib==3.4.3
    - panel==0.12.4
    - voila==0.2.16
    - streamlit==1.0.0
    - dash==2.0.0
    - cdsdashboards-singleuser==0.6.0
```

### Command output

```bash session
INFO:qhub.provider.terraform:terraform init took 6.004 [s]
INFO:qhub.provider.terraform:terraform apply directory=stages/03-kubernetes-initialize targets=[]
INFO:qhub.provider.terraform: terraform at /tmp/terraform/1.0.5/terraform
[terraform]: Acquiring state lock. This may take a few moments...
[terraform]: ╷
[terraform]: │ Error: error reading EKS Cluster (qhubdemo-dev): couldn't find resource
[terraform]: │ 
[terraform]: │   with data.aws_eks_cluster.default,
[terraform]: │   on _qhub.tf.json line 28, in data.aws_eks_cluster.default:
[terraform]: │   28:             }
[terraform]: │ 
[terraform]: ╵
[terraform]: Releasing state lock. This may take a few moments...

Problem encountered: Terraform error
```

### Versions and dependencies used.

```
$ conda --version
conda 4.11.0
```

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T16:31:21Z", GoVersion:"go1.16.1", Compiler:"gc", Platform:"linux/amd64"}
error: You must be logged in to the server (the server has asked for the client to provide credentials)
```

```
$ qhub --version
0.4.0
```

### PR DESCRIPTION
**Pull request by tylerpotts**
Fixes | Closes | Resolves #1228

## Changes introduced in this PR:

- Adds to docs regarding `AWS_DEFAULT_REGION` env var
- Adds check for presence of variable in deploy scripts
## Types of changes

What types of changes does your PR introduce?

_Put an `x` in the boxes that apply_

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features to not work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

### CODE REVIEW

**Comment by danlester** [comment]
I think we went in the other direction on this a few months ago...

The issue is that the region is also a key in the qhub-config.yaml file, so which is the 'right' one? And does the terraform provider or anything else respect the env var where the rest of our code is looking at the yaml file?

I think last time around we took a look at the env var during `qhub init` to override the default output into the qhub-config.yaml if the env var is set, and attempted to ignore the env var from then onwards...

Maybe we even need to insist the env var is NOT set so that it doesn't cause any confusion during `qhub deploy`!

**Comment by costrouc** [comment]
@danlester I get what you mean and agree. From a reproducibility standpoint I think forcing the region to be specified in the qhub-config.yaml is preferable. 

I would prefer if we check that this variable is not set.

**Comment by tylerpotts** [comment]
@costrouc @danlester That sounds reasonable to me.

The current issue is that running `qhub deploy` picks up the region correctly from `qhub-config.yaml` for some of the deployment, but for other parts grabs the default region that is configured in `~/.aws/config`. This can lead to errors when terraform attempts to find resources in regions where they don't exist:

```
[terraform]: │ Error: error configuring Terraform AWS Provider: error validating provider credentials: error calling sts:GetCallerIdentity: operation error STS: GetCallerIdentity, failed to resolve service endpoint, an AWS region is required, but was not found
```
Complete log here: https://github.com/Quansight/qhub/issues/1228#issuecomment-1090419813

File: docs/source/installation/setup.md [hunk 1]
```diff
@@ -63,11 +63,13 @@ For more details on configuration for each Cloud provider, check the How-To Guid
 Please see these instructions for [creating an IAM role](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create.html) with administrator permissions. Upon generation, the
 IAM role will provide a public **access key ID** and a **secret key** which will need to be added to the environment variables.
 
-To define the environment variables paste the commands below with your respective keys.
+To define the environment variables paste the commands below with your respective keys. The default region generated by `qhub init aws ...` will be `us-west-2`
 
 ```shell
 export AWS_ACCESS_KEY_ID="HAKUNAMATATA"
 export AWS_SECRET_ACCESS_KEY="iNtheJUng1etheMightyJUNgleTHEl10N51eEpsT0n1ghy;"
+export
```
[the remark this answers is not in the document]
Comment (tylerpotts) [reply docs/source/installation/setup.md:71]: Yes... Failed a copy/paste

**Comment by costrouc** [comment]
@viniciusdc please open an issue regarding this PR. @tylerpotts we brought this up in the meeting today that we need to be more explicit in the terraform modules which region is being used in the AWS provider.

**Comment by viniciusdc** [comment]
Hi @tylerpotts, we actually found a bug with the terraform state files that we had for AWS that generated this weird behavior with `AWS_REGION` #1271. I will be closing this PR due to that, please let me know if that fixes the error you were seen.

**Pull request closed without merging**